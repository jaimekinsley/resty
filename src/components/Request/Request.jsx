import React from 'react';
import PropTypes from 'prop-types';

const Request = ({ method, api }) => (
  <>
    <p>{method}</p>
    <p>{api}</p>
  </>
);

Request.propTypes = {
  method: PropTypes.string.isRequired,
  api: PropTypes.string.isRequired
};

export default Request;
