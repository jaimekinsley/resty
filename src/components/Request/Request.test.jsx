import React from 'react';
import { shallow } from 'enzyme';
import Request from './Request';

describe('Request component', () => {
  it('renders a Request', () => {
    const wrapper = shallow(<Request method="get" api="https://futuramaapi.com" />);
    expect(wrapper).toMatchSnapshot();
  });
});
