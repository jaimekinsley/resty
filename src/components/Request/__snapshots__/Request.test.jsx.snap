// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request component renders a Request 1`] = `
<Fragment>
  <p>
    get
  </p>
  <p>
    https://futuramaapi.com
  </p>
</Fragment>
`;
